# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Caches-inl.h
# Commit: 9bfa6023a553
# Full Hash: 9bfa6023a5539598e824e3d2a95dec25d099a378
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 5 - Remove JSClass* from ObjectGroup. r=jonco
#   
#   BaseShape stores the JSClass* too so this patch changes the code to get the JSClass*
#   from there instead of from the ObjectGroup.
#   
# ==============================================================================

diff -r 2186bd51bebd -r 9bfa6023a553 js/src/vm/Caches-inl.h
--- a/js/src/vm/Caches-inl.h	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/vm/Caches-inl.h	Sat Mar 06 01:05:20 2021 +0000
@@ -62,15 +62,16 @@
     return nullptr;
   }
 
+  const JSClass* clasp = templateObj->getClass();
   NativeObject* obj = static_cast<NativeObject*>(AllocateObject<NoGC>(
-      cx, entry->kind, /* nDynamicSlots = */ 0, heap, group->clasp()));
+      cx, entry->kind, /* nDynamicSlots = */ 0, heap, clasp));
   if (!obj) {
     return nullptr;
   }
 
   copyCachedToObject(obj, templateObj, entry->kind);
 
-  if (group->clasp()->shouldDelayMetadataBuilder()) {
+  if (clasp->shouldDelayMetadataBuilder()) {
     cx->realm()->setObjectPendingMetadata(cx, obj);
   } else {
     obj = static_cast<NativeObject*>(SetNewObjectMetadata(cx, obj));